sql/src/main/java/io/crate/execution/dml/upsert/CheckConstraints.java

```diff
@@ -71,5 +78,22 @@ public void validate(T values) {
                 throw new IllegalArgumentException("\"" + notNullColumns.get(i) + "\" must not be null");
             }
         }
+        for (int i = 0; i < checkConstraints.size(); i++) {
+            Tuple<? extends Input<?>, CheckConstraint<Symbol>> checkEntry = checkConstraints.get(i);
+            Input<?> checkInput = checkEntry.v1();
+            Boolean value = (Boolean) checkInput.value();
```

Comment:
maybe worth adding an assertion to ensure its a boolean? Is this ensured somewhere else? otherwise it should throw an exception here.

Reply:
the grammar makes it be the result of a boolean expression, so it will always be boolean

Reply:
I think it's worth adding an assertion or inline comment that clarifies that

Reply:
Yes lets please add at least a comment, I'd even add an assertion.

Reply:
Assertion added.